controller/src/beerocks/master/db/node.h

```diff
@@ -211,6 +212,29 @@ class node {
     std::unordered_map<sMacAddr, std::unordered_map<sMacAddr, link_metrics_data>>
         m_link_metric_data;
 
+    class ap_metrics_data {
+    public:
+        ap_metrics_data(){};
+        ~ap_metrics_data(){};
+
+        sMacAddr bssid;
+        uint8_t channel_utilization;
+        uint16_t number_of_stas_currently_associated;
+        uint8_t estimated_service_info_field_ac_be[3];
+        uint8_t estimated_service_info_field_ac_bk[3];
+        uint8_t estimated_service_info_field_ac_vo[3];
+        uint8_t estimated_service_info_field_ac_vi[3];
+
+        bool add_ap_metric_data(std::shared_ptr<wfa_map::tlvApMetric> ApMetricData);
+    };
+
+    /*
+    * This map holds ap metric data per reporting Agent sMacAddr .
+    * Map is Used in TYPE_GW/TYPE_IRE nodes.
+    * Map created empty in all other nodes.
+    */
+    std::unordered_map<sMacAddr, ap_metrics_data> m_ap_metric_data;
```

Comment:
This map shouldn't be part of the GW/IRE node. Instead, it should be part of the node that corresponds to a BSS. Unfortunately, the db doesn't currently model it that way - we do have son::node::radio but no per-BSS (i.e. per-VAP) info. Actually, that is probably a mistake, because there is already the stats_info which contains the per-BSS stats.

In fact, probably you should just update ap_stats_params with the service info. active_sta_count and channel_load_percent are already there.

Alternatively, I'm OK with adding an issue to clean this up later.